benches/benches/vm.rs

```diff
@@ -37,15 +39,17 @@ where
 
             let clock = quanta::Clock::new();
 
+            let original_db = vm.as_mut().database_mut().clone();
+            // Simulates the block production/validation with three levels of database transaction.
+            let block_database_tx = original_db.clone().into_transaction();
+            let relayer_database_tx = block_database_tx.into_transaction();
+            let thread_database_tx = relayer_database_tx.into_transaction();
+            let tx_database_tx = thread_database_tx.into_transaction();
```

Comment:
I'm not sure what's going on here. You seem to be wrapping each in another `StorageTransaction`, but I don't know how that relates to `block`, `relayer`, `thread`, etc.

Reply:
Added a comment with more detailed description=)